patches/0005-LPS-112982-Add-additional-resource-URL-parameters.patch

```diff
@@ -0,0 +1,71 @@
+From 1551d5d3ab5926fcf22e78837da2fac279b6177b Mon Sep 17 00:00:00 2001
+From: Julien Castelain <julien.castelain@liferay.com>
+Date: Mon, 6 Jul 2020 11:55:54 +0200
+Subject: [PATCH 5/5] LPS-112982 Add additional resource URL parameters
+
+---
+ core/ckeditor_base.js | 25 ++++++++++++++++++-------
+ core/config.js        |  9 +++++++++
+ 2 files changed, 27 insertions(+), 7 deletions(-)
+
+diff --git a/core/ckeditor_base.js b/core/ckeditor_base.js
+index 3fb18f5f7..e283f2011 100644
+--- a/core/ckeditor_base.js
++++ b/core/ckeditor_base.js
+@@ -171,15 +171,26 @@ if ( !window.CKEDITOR ) {
+ 			 * @returns {String} The full URL.
+ 			 */
+ 			getUrl: function( resource ) {
+-				// If this is not a full or absolute path.
+-				if ( resource.indexOf( ':/' ) == -1 && resource.indexOf( '/' ) !== 0 )
+-					resource = this.basePath + resource;
++				var resourceUrl = resource.includes(CKEDITOR.basePath)
++						? new URL(resource)
++						: new URL(CKEDITOR.basePath + resource);
+ 
+-				// Add the timestamp, except for directories.
+-				if ( this.timestamp && resource.charAt( resource.length - 1 ) != '/' && !( /[&?]t=/ ).test( resource ) )
+-					resource += ( resource.indexOf( '?' ) >= 0 ? '&' : '?' ) + 't=' + this.timestamp;
++				if (!resourceUrl.toString().endsWith('/')) {
++					if (!resourceUrl.searchParams.has('t')) {
++						resourceUrl.searchParams.set('t', CKEDITOR.timestamp);
++					}
++
++					Object.entries(CKEDITOR.ADDITIONAL_RESOURCE_PARAMS).forEach(function(entries) {
```

Comment:
What if `CKEDITOR.ADDITIONAL_RESOURCE_PARAMS` is `null`?

Reply:
Also, I think you can do, `function([key, value])`

Reply:
Also, the reason I offered 2 options in my previous comment is to make sure we don't run into any race conditions where we set `CKEDITOR.ADDITIONAL_RESOURCE_PARAMS` **AFTER** we create URLs.

Can you confirm that won't be the case here or provide some insight into why you chose this option?

Reply:
It's always defined as an empty object, but if someone overrides it by doing something like 

`CKEDITOR.ADDITIONAL_RESOURCE_PARAMS = null;`

or something similar, it would be better to check :+1:

Reply:
> Also, I think you can do, `function([key, value])`

Sure about that? This won't be transpiled, so IE [would choke on it](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment).

Still ok to use `Object.entries` though due to polyfill in DXP.

Reply:
Ugh, forget about it, then...

Also, I didn't see https://github.com/liferay/liferay-ckeditor/pull/89/files#diff-22c7e98f6759e5b24ef2a7f9298289aaR64

I'll leave the reviews to you, @wincent!

Reply:
Yes, unfortunately, we can't use destructuring, but it would have been nice

Reply:
> I'll leave the reviews to you, @wincent!

But, but... [Linus's Law](https://en.wikipedia.org/wiki/Linus%27s_law)!

> given enough eyeballs, all bugs are shallow

Reply:
> Also, I didn't see https://github.com/liferay/liferay-ckeditor/pull/89/files#diff-22c7e98f6759e5b24ef2a7f9298289aaR64

No problem, it's actually a good idea, because even though I don't know why someone would do it, they still can do 

`CKEDITOR.ADDITIONAL_RESOURCE_PARAMS = null;`

... but should we cover all cases?
 
What if someone does this

`CKEDITOR.ADDITIONAL_RESOURCE_PARAMS = "gotcha";`

We'd still need to check that `CKEDITOR.ADDITIONAL_RESOURCE_PARAMS` is non null, and is an object, but since I know we don't want to be defensive, I'm thinking that if someone changes this to `null` or anything else than an object and it blows up they kind of get what they asked for ... but maybe that's a bit harsh